wifinina/tcp.go

```diff
@@ -32,7 +60,97 @@ type readBuffer struct {
 	size int
 }
 
+type accessPoint struct {
+	ssid     string
+	password string
+	timeout  time.Duration
+}
+
+// HandleWifi ensures device is connected to an access point (if any) and is called before any network operation
+func (drv *Driver) HandleWifi() error {
```

Comment:
Why not name this `EnsureConnected()`?

Also, the first thing it should do is:

```go
if !drv.autoReconnect {
   return nil
}
```

This way only if the user has set `device.UseAutoReconnect(true)` that this is used.